src/components/organisms/ChangePasswordModal/ChangePasswordModal.tsx

```diff
@@ -1,18 +1,117 @@
-import React from "react";
+import React, { useState } from "react";
 import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import "./ChangePasswordModal.scss";
+import { useFirebase } from "react-redux-firebase";
 
 interface PropsType {
   show: boolean;
   onHide: () => void;
 }
 
+interface ChangePasswordData {
+  currentPassword: string;
+  newPassword: string;
+  confirmNewPassword: string;
+}
+
 const ChangePasswordModal: React.FunctionComponent<PropsType> = ({
   show,
   onHide,
-}) => (
-  <Modal show={show} onHide={onHide}>
-    <Modal.Body>Hello World</Modal.Body>
-  </Modal>
-);
+}) => {
+  const firebase = useFirebase();
+  const user = firebase.auth().currentUser;
+  const { register, handleSubmit, errors, formState, watch } = useForm<
+    ChangePasswordData
+  >({
+    mode: "onChange",
+  });
+
+  const [incorrectCurrentPassword, setIncorrectCurrentPassword] = useState(
+    false
+  );
+
+  const onSubmit = async (data: ChangePasswordData) => {
+    try {
+      user &&
```

Comment:
Can you do all these checks out of the function, and return early if they're missing?